papermill/iorw.py

```diff
@@ -88,6 +88,11 @@ def listdir(cls, path):
 
     @classmethod
     def write(cls, buf, path):
+        dir = os.path.dirname(path)
```

Comment:
Since `dir` is a reserved keyword in python let's use a different name here, or just combine this with the last line to get `if not os.path.exists(os.path.dirname(path)):`

Reply:
@MSeal removed the dir and used os.path.dir(path)